coalib/settings/ConfigurationGathering.py

```diff
@@ -262,7 +262,9 @@ def get_config_directory(section):
 
     However if its origin is already a directory this will be preserved:
 
-    >>> section['files'].origin = os.path.abspath('/tmp/dir/')
+    >>> files = section['files']
+    >>> files.origin = os.path.abspath('/tmp/dir/')
+    >>> section.append(files)
```

Comment:
https://github.com/coala/coala/pull/3178/commits/2df2936b7f6448c568b6252304c1c6e90d65742e#diff-41b4f7bdbd660f8c693c7d0304602d79R205 . Because of this here it doesn't allow changing the settings within a Section without replacing them.